src/org/zaproxy/zap/extension/ascanrulesAlpha/IntegerOverflow.java

```diff
@@ -0,0 +1,271 @@
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpResponseHeader;
+import org.parosproxy.paros.network.HttpStatusCode;
+
+
+public class IntegerOverflow extends AbstractAppParamPlugin  {
+
+	/**
+	 * Prefix for internationalised messages used by this rule
+	 */
+	private static final String MESSAGE_PREFIX = "ascanalpha.integeroverflow.";
+	private static final int PLUGIN_ID = 30003;
+	private static Logger log = Logger.getLogger(IntegerOverflow.class);
+	
+	@Override
+	public int getId() {
+		return PLUGIN_ID;
+	}
+
+	@Override
+	public String getName() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "name");
+	}
+
+	@Override
+	public String[] getDependency() {
+		return null;
+	}
+
+	@Override
+	public String getDescription() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "desc");
+	}
+
+	@Override
+	public int getCategory() {
+		return Category.INJECTION;
+	}
+
+	@Override
+	public String getSolution() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "soln");
+	}
+
+	@Override
+	public String getReference() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "refs");
+	}
+	
+	public String getOther() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "other");
+	}
+	
+	private String getError(char c) {
+		return Constant.messages.getString(MESSAGE_PREFIX + "error"+c);
+	}
+	
+	@Override
+	public void init() {
+
+	}
+
+	/*
+	 * This method is called by the active scanner for each GET and POST parameter for every page 
+	 * @see org.parosproxy.paros.core.scanner.AbstractAppParamPlugin#scan(org.parosproxy.paros.network.HttpMessage, java.lang.String, java.lang.String)
+	 */
+	@Override
+	public void scan(HttpMessage msg, String param, String value) {
+	
+		if (this.isStop()) { // Check if the user stopped things
+			if (log.isDebugEnabled()) {
+				log.debug("Scanner "+this.getName()+" Stopping.");
+			}
+			return; // Stop!
+		}
+		
+		try {
+			
+			msg = getNewMsg();
+			String returnAttack = randomIntegerString(4);// The number of full length ints to send.
+			setParameter(msg, param, returnAttack);
+			sendAndReceive(msg);
+			HttpResponseHeader requestReturn = msg.getResponseHeader();			
+			// This is where BASE baseResponseBody was you detect potential vulnerabilities in the response
+    		String chkerrorheader = requestReturn.getHeadersAsString();
+    		log.debug("Header: "+ chkerrorheader);
+    		if (msg.getResponseHeader().getStatusCode() == HttpStatusCode.INTERNAL_SERVER_ERROR )
+    		{
+    			log.debug("Found Header");
+    			bingo(getRisk(), 
+    					Alert.CONFIDENCE_MEDIUM, 
+    					this.getBaseMsg().getRequestHeader().getURI().toString(), 
+    					param, 
+    					msg.getRequestHeader().toString(), 
+    					this.getError('1') ,
+    					msg);
+    			return;
+    		}
+    		msg = getNewMsg();

```

Comment:
Consider checking again if the scan has been stopped before continuing the following test (the same comment for the other tests).

Since it's basically doing the same things for every attack, it could/should be extract a method, that performs the injection, sends the message and raises the alert. A method with following method signature?
`boolean attack(String param, String attack, char otherInfoKey) throws IOException`